### PR TITLE
Display CPhaseGate as symmetric in all circuit drawers

### LINKED ISSUE #5782: controlled-phase gate (CPhase) should draw symmetrically
Instead of this:
image

It should show like this, because it is symmetric (similar to CU1, RZZ, etc.)
image

This applies to all circuit drawers.

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fixes #5782

### Details and comments

This PR changes the display of the `CPhaseGate `or `cp `gate to display as a symmetric controlled gate in the mpl, text, and latex drawers, the same way as the `cu1 `gate. As part of this, the 'u1' name was added to the display of the `cu1 `gate in the text and latex drawer for consistency and to differentiate it from the `cp `gate.
```
q_0: ─■─────────■───────
      │u1(π/2)  │p(π/2) 
q_1: ─■─────────■───────
```

### CODE REVIEW

File: qiskit/visualization/matplotlib.py [hunk 1]
```diff
@@ -825,8 +825,14 @@ def _draw_ops(self, verbose=False):
                 else:
                     param_width = 0.0
 
-                if op.name == 'cu1' or op.name == 'rzz' or base_name == 'rzz':
-                    tname = 'U1' if op.name == 'cu1' else 'zz'
+                if (op.name == 'cu1' or op.name == 'cp'
+                        or op.name == 'rzz' or base_name == 'rzz'):
```
Comment (Cryoris) [inline qiskit/visualization/matplotlib.py:829]:
```suggestion
                if op.name in ['cu1', 'cp', 'rzz'] or base_name == 'rzz':
```

File: qiskit/visualization/matplotlib.py [hunk 2]
```diff
@@ -990,16 +996,22 @@ def _draw_ops(self, verbose=False):
                     self._ctrl_qubit(q_xy[1], fc=ec, ec=ec, tc=tc)
                     self._line(qreg_b, qreg_t, lc=lc, zorder=PORDER_LINE + 1)
 
-                # cu1, rzz, and controlled rzz gates (sidetext gates)
-                elif (op.name == 'cu1' or op.name == 'rzz' or base_name == 'rzz'):
+                # cu1, cp, rzz, and controlled rzz gates (sidetext gates)
+                elif (op.name == 'cu1' or op.name == 'cp'
+                      or op.name == 'rzz' or base_name == 'rzz'):
```
Comment (Cryoris) [inline qiskit/visualization/matplotlib.py:1001]:
```suggestion
                elif op.name in ['cu1', 'cp', 'rzz'] or base_name == 'rzz':
```

File: qiskit/visualization/matplotlib.py [hunk 3]
```diff
@@ -990,16 +996,22 @@ def _draw_ops(self, verbose=False):
...
                     num_ctrl_qubits = 0 if op.name == 'rzz' else op.op.num_ctrl_qubits
                     if op.name != 'rzz':
                         self._set_ctrl_bits(op.op.ctrl_state, num_ctrl_qubits,
                                             q_xy, ec=ec, tc=tc, text=ctrl_text, qargs=op.qargs)
                     self._ctrl_qubit(q_xy[num_ctrl_qubits], fc=ec, ec=ec, tc=tc)
-                    if op.name != 'cu1':
+                    if op.name != 'cu1' and op.name != 'cp':
```
Comment (Cryoris) [inline qiskit/visualization/matplotlib.py:1007]:
```suggestion
                    if op.name not in ['cu1', 'cp']:
```

**Comment by Cryoris** [comment]
I think the approach looks good, though I'm not familiar with the latex part. Do we also have tests for that?

**Comment by enavarro51** [comment]
To be honest (hope I don't offend anyone here), the latex drawer is a bit of a mess and the testing for it is seriously inadequate. I did the minimum needed here to deal with cu1 and cp (though I did fix zz, which was printing the label a full wire below the gate). Even coming up with tests for this is difficult. For example,
```
qc = QuantumCircuit(2)
qc.cu1(pi/2, 0, 1)
qc.cp(pi/2, 0, 1)
qc.draw('latex')
```
image
```
qc = QuantumCircuit(2)
qc.x(0)
qc.cu1(pi/2, 0, 1)
qc.cp(pi/2, 0, 1)
qc.draw('latex')
```
image
I've run test circuits on the latex drawer and could come up with a dozen or more issues from those. 
So for this PR, I think we can go with what is, and hopefully some day the latex drawer will get the love it needs,.

File: qiskit/visualization/matplotlib.py (hunk 1 above)
Comment (enavarro51) [reply qiskit/visualization/matplotlib.py]: Done in https://github.com/Qiskit/qiskit-terra/pull/5804/commits/54073ddf01439a371ed126e03296a4c61e9bffa3

File: qiskit/visualization/matplotlib.py (hunk 2 above)
Comment (enavarro51) [reply qiskit/visualization/matplotlib.py]: Done in https://github.com/Qiskit/qiskit-terra/pull/5804/commits/54073ddf01439a371ed126e03296a4c61e9bffa3

File: qiskit/visualization/matplotlib.py (hunk 3 above)
Comment (enavarro51) [reply qiskit/visualization/matplotlib.py]: Done in https://github.com/Qiskit/qiskit-terra/pull/5804/commits/54073ddf01439a371ed126e03296a4c61e9bffa3

**Comment by Cryoris** [comment]
I have a few comments but I think this is mostly ready to merge.

1. Can we make the gate labels on the symmetric gates all uppercase to match with the names of the boxed gates? Right now they are `u1 p zz` but I think they should be `U1 P ZZ`
1. There are two new empty files, are these supposed to be there?
     
     ```
     test/ipynb/mpl/references/iqx_color.png
     test/ipynb/mpl/references/user_style.png
     ```
1. In a larger circuit, the string `P(angle)` is on top of the control wire for me. Is this something we can fix?
image

File: qiskit/visualization/latex.py [hunk 4]
```diff
@@ -292,8 +292,8 @@ def _get_image_depth(self):
         for layer in self.ops:
             column_width = 1
             for nd in layer:
-                if nd.name in ['cu1', 'rzz']:
-                    column_width = 2
+                if nd.name in ['cu1', 'cp', 'rzz']:
+                    column_width = 3
```
Comment (1ucian0) [inline qiskit/visualization/latex.py:296]: why?

File: qiskit/visualization/latex.py [hunk 5]
```diff
@@ -747,18 +752,20 @@ def _build_latex_array(self):
                                         "\\ctrl{" + str(pos_2 - pos_1) + "}"
                                 self._latex[pos_2][column] = \
                                     "\\gate{R_z(%s)}" % (self.parse_params(op.op.params[0]))
-                            elif nm == "cu1":
+                            elif nm in ['cu1', 'cp']:
```
Comment (1ucian0) [inline qiskit/visualization/latex.py:755]: what about `'rzz'`?

File: qiskit/visualization/latex.py (hunk 4 above)
Comment (enavarro51) [reply qiskit/visualization/latex.py:296]: Previously we only showed the phase for 'u1', such as (pi/2). Now we add the name. Below is =3 on the left, =2 on the right.
image

File: qiskit/visualization/latex.py (hunk 5 above)
Comment (enavarro51) [reply qiskit/visualization/latex.py:755]: In this never-ending 'elif', 'rzz is handled below at line 781. All of this gets cleaned up on #5855, of course. :-)

**Comment by enavarro51** [comment]
@Cryoris,
1. This is for the 'text' drawer. The names are uppercase in the 'mpl' and 'latex' drawers. I agree they should all be consistent, so I'll make them upper in the 'text' drawer. For some reason 'zz' has always been lower case in all the drawers. Should it go upper as well?
2. These are actually not empty, but are reference files for the 'mpl' tests. These 2 images needed to be changed because they had symmetric gates in them.
3. l can't recreate this problem. I've run it on jupyter and using command line linux and I get this. This may be a different matlplotlib backend issue. I can move it over just a bit to cover your case, but I wouldn't be able to test it.
image

**Comment by enavarro51** [comment]
I believe everything is covered in https://github.com/Qiskit/qiskit-terra/pull/5804/commits/73ed3b6b3ec2cabac26fea9cc3878a0f46e2f9c2.

**Review by Cryoris — APPROVED** [review]
LGTM thanks for updates!

**Review by 1ucian0 — APPROVED** [review]

**Pull request merged**